lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -104,30 +104,51 @@ def self.bds
     # idempotency issue as system add command throws error if a bd is already
     # present in the system range.
     def create
-      sys_bds_array = BridgeDomain.bd_ids_to_array(system_bd_add)
+      sys_bds_array = BridgeDomain.bd_ids_to_array(system_bridge_domain)
       inp_bds_array = BridgeDomain.bd_ids_to_array(@bd_ids)
       if (inp_bds_array - sys_bds_array).any?
         add_bds = BridgeDomain.bd_list_to_string(inp_bds_array - sys_bds_array)
-        config_set('bridge_domain', 'system_bd_add', addbd: add_bds)
+        config_set('bridge_domain', 'system_bridge_domain', oper: 'add',
+                                                            bd:   add_bds)
       end
-      config_set('bridge_domain', 'create', crbd: @bd_ids)
+      config_set('bridge_domain', 'create', bd: @bd_ids)
     end
 
     def destroy
-      config_set('bridge_domain', 'destroy', delbd: @bd_ids, rembd: @bd_ids)
+      config_set('bridge_domain', 'destroy', bd: @bd_ids)
+      config_set('bridge_domain', 'system_bridge_domain', oper: 'remove',
+                                                          bd:   @bd_ids)
     end
 
     ########################################################
     #                      PROPERTIES                      #
     ########################################################
 
+    # Bridge-Domain name assigning case
+    # bridge-domain 100
+    #   name PepsiCo
+    def bd_name
+      res = config_get('bridge_domain', 'bd_name', bd: @bd_ids)
+      res.nil? ? default_bd_name : res
+    end
```

Comment:
This is better but you shouldn't need the nil check since default_bd_name returns nil. All you should need here is:

```
    def bd_name
      config_get('bridge_domain', 'bd_name', bd: @bd_ids)
    end
```